dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/DefaultAnalyticalObjectImportHandler.java

```diff
@@ -241,6 +242,18 @@ private void handleDataDimensionItems(
                         dataDimensionItem.getProgramAttribute().getAttribute()));
       }
 
+      // handle expression dimension item
+      ExpressionDimensionItem expressionDimensionItem =
+          dataDimensionItem.getExpressionDimensionItem();
+      if (expressionDimensionItem != null) {
+        ExpressionDimensionItem expressionDimensionItemBundle =
+            bundle.getPreheat().get(bundle.getPreheatIdentifier(), expressionDimensionItem);
+        // use bundle object if available to avoid transient exception
+        if (expressionDimensionItemBundle != null) {
+          dataDimensionItem.setExpressionDimensionItem(expressionDimensionItemBundle);
+        }
+      }
+
```

Comment:
I don't have the full picture here. I see that this change (and also for the test file below) comes from david-mackessy (https://github.com/EyeSeeTea/dhis2-core/pull/95/commits/f37846764c77c306db06ceb678216ea6f41146ae).

It may be related to our problem but it's not evident. Why do we need it? Maybe it's documented in the clickup task or PR description but I can't find it.

Reply:
Ah, I see, it's #94. Let's keep them isolated, using 2.40.4.1 as base branch for all PRs. This way the diff shows exactly what we are doing for that particular task.